src/Workspaces/CSharp/Portable/Extensions/ITypeParameterSymbolExtensions.cs

```diff
@@ -42,6 +42,10 @@ internal static class ITypeParameterSymbolExtensions
             {
                 constraints.Add(SyntaxFactory.ClassOrStructConstraint(SyntaxKind.ClassConstraint));
             }
+            else if (typeParameter.HasUnmanagedTypeConstraint)
+            {
+                constraints.Add(SyntaxFactory.TypeConstraint(SyntaxFactory.IdentifierName("unmanaged")));
```

Comment:
Should this be SyntaxFactory.ClassOrStruct constraint? Or should there be a new syntax factory method? I guess I think of this as analogous to the keyword ones, hence wondering if it fits there.

Reply:
It should generate the same code as what was parsed out IMO.  So this should be a "TypeConstraint" with "unmanaged" as the type name.  just like "var" is parsed out just an type with an identifier name.  It's only semantic analysis that imbues these identifiers with special meaning.